google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -73,7 +73,7 @@
   private static final String MANIFEST_ARTIFACT_ID_KEY = "artifactId";
   private static final String MANIFEST_VERSION_KEY = "Implementation-Version";
   private static final String ARTIFACT_ID = "google-cloud-core";
-  private static final String LIBRARY_NAME = "gcloud-java";
```

Comment:
As confirmed offline: `GrpcServiceOptions` sets the headers we expect (`x-goog-api-client`), but changing `LIBRARY_NAME` also affects the `User-Agent` value for non-gRPC clients. For now, let's separate the value used for gRPC and non-gRPC.